modules/backup-s3/client.go

```diff
@@ -199,17 +258,31 @@ func (s *s3Client) Write(ctx context.Context, backupID, key string, r io.ReadClo
 	return info.Size, nil
 }
 
-func (s *s3Client) Read(ctx context.Context, backupID, key string, w io.WriteCloser) (int64, error) {
+func (s *s3Client) Read(ctx context.Context, backupID, key, overrideBucket, overridePath string, w io.WriteCloser) (int64, error) {
 	defer w.Close()
-	path := s.makeObjectName(backupID, key)
-	obj, err := s.client.GetObject(ctx, s.config.Bucket, path, minio.GetObjectOptions{})
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return -1, errors.Wrap(err, "GetObject")
```

Comment:
suggestin: maybe `read: cannot get client` would be a better message?

Reply:
done